provisioning/setup.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import asyncio
+import argparse
+import json
+import logging
+from subprocess import Popen, PIPE
+import shutil
+from eos.utils.product_features import unsupported_features
+
+
+def get_data_from_provisioner_cli(method, output_format="json") -> str:
+    rc = 1
+    try:
+        process = Popen(['provisioner', f"{method}", f"--out={output_format}"],
+                        stdout=PIPE, stderr=PIPE)
+        process.wait()
+        stdout, err = process.communicate()
+        rc = process.returncode
+    except Exception as e:
+        logging.error('Failed to fetch data from prvosioner (%s)', e)
+    if rc == 0:
+        res = stdout.decode('utf-8')
+        if res != "":
+            result: str = json.loads(res)['ret']
+            return result
+    return 'unknown'
+
+
+def logrotate_config():
+    try:
+        setup_info = get_data_from_provisioner_cli('get_setup_info')
+        setup_type = setup_info['server_type']
+        shutil.copyfile(
+            f"/opt/seagate/cortx/hare/conf/logrotate/{setup_type}",
+	     '/etc/logrotate.d/hare')
+    except Exception as error:
+        logging.error('Error setting logrotate values for hare (%s)', error)
+
+def _report_unsupported_features(features_unavailable):
+    async def _process(features_unavailable):
+        loop = asyncio.get_event_loop()
+        uf_db = await loop.run_in_executor(None,
+                    unsupported_features.UnsupportedFeaturesDB)
+        loop = asyncio.get_event_loop()
+        await loop.run_in_executor(None, uf_db.store_unsupported_features,
+                                       str('hare'), features_unavailable)
+    _process
```

Comment:
I believe it should be `return _process` instead

Reply:
I tried that actually, so I added a log in this function which does not get posted so I am not sure if the function is getting executed or not.

Reply:
I've just added a commit to your branch. I had no ability to check it, but probably it works better now.